blaze/server/server.py

```diff
@@ -316,14 +317,19 @@ def from_tree(expr, namespace=None):
         return expr
 
 
-@api.route('/compute.<serial_format>',
-           methods=['POST', 'HEAD', 'OPTIONS'])
+mimetype_regex = re.compile(r'^application/vnd\.blaze\+(%s)$' %
+                            '|'.join(x.name for x in all_formats))
+
+
+@api.route('/compute', methods=['POST', 'HEAD', 'OPTIONS'])
 @crossdomain(origin='*', methods=['POST', 'HEAD', 'OPTIONS'])
-def compserver(serial_format):
+def compserver():
+    content_type = request.headers['content-type']
+
     try:
-        serial = _get_format(serial_format)
+        serial = _get_format(*mimetype_regex.match(content_type).groups())
```

Comment:
The * makes me think that this could possibly be more than one value; could we just take [0] instead?